code/obj/item/building_materials.dm

```diff
@@ -271,7 +271,8 @@ MATERIAL
 			ui.open()
 
 	attack_self(mob/user as mob)
-		ui_interact(user)
+		if (isturf(user.loc))
+			ui_interact(user)
```

Comment:
what's wrong with making rods in a locker?

Reply:
#16271, You can build things such as a Display Case in the gene booth, blocking other players from using it.

Reply:
Resolved, you can now build non-dense things in things (Rods in a locker is fine now)